### ISSUE TITLE
Well organized documentation for QuISP

### ISSUE THREAD

**Issue by wolfitis** [sfc-aqua/quisp#172]
There are many markdown files in /doc directory that are not easy to follow for people new to QuISP

A well formatted and organized documentation will be a good reference for anyone new to this package. Addition of some basic easy to follow tutorials can be a good starting point for beginners.

To arrange the available docs in some order and link them all together.

Many open source projects have amazing documentations that can be taken as example.

**Comment by zigen** [comment]
how about this structure? 
I think one of the problems is the documentation is far from the actual corresponding source code.
The documents that describe the implementation should be put near the source code.
For me, it's better than putting whole documents in `./doc` directory.
```
.
├── Authors.md
├── bin
├── dict.txt
├── doc
├── Dockerfile
├── Doxyfile
├── Makefile
├── quisp
│   ├── channels
│   │   └── QuantumChannel.cc
│   ├── classical_messages_m.cc
│   ├── classical_messages_m.h
│   ├── classical_messages.msg
│   ├── Makefile
│   ├── makemakefiles
│   ├── modules
│   │   ├── Application
│   │   │   ├── Application.cc
│   │   │   ├── Application.h
│   │   │   └── README.md
│   │   ├── QRSA
│   │   │   ├── ConnectionManager
│   │   │   │   ├── ConnectionManager.cc
│   │   │   │   ├── ConnectionManager.h
│   │   │   │   └── README.md
│   │   │   ├── HardwareMonitor
│   │   │   │   ├── HardwareMonitor.cc
│   │   │   │   ├── HardwareMonitor.h
│   │   │   │   └── README.md
│   │   │   ├── RealTimeController
│   │   │   │   ├── RealTimeController.cc
│   │   │   │   ├── RealTimeController.h
│   │   │   │   └── README.md
│   │   │   ├── QRSA.ned
│   │   │   └── README.md
│   │   ├── HoM
│   │   │   ├── BellStateAnalyzer
│   │   │   │   ├── BellStateAnalyzer.cc
│   │   │   │   ├── BellStateAnalyzer.h
│   │   │   │   └── README.md
│   │   │   ├── HoMController
│   │   │   │   ├── HoMController.cc
│   │   │   │   ├── HoMController.h
│   │   │   │   └── README.md
│   │   │   ├── HoM.ned
│   │   │   └── README.md
```

**Comment by icepolarizer** [comment]
How about using [readthedocs.org](https://readthedocs.org/) ? If somebody can assign this to me, I'll do the job.

**Comment by rdviii** [comment]
I'm not familiar with readthedocs.org. Having something that indexes and creates a table of contents and turns the various markdown files into a single PDF would be useful.

**Comment by rdviii** [comment]
@zigen your proposal above looks like it's primarily about revising the subdirectory structure for the source, not the documentation?

**Comment by rdviii** [comment]
Documentation, of course, has multiple purposes.  A bunch of what's there so far is either old _design_ docs, some dating back several years, or a _narrative_/tutorial for beginners, to get them started.  Sounds like what @zigen wants is documentation to accompany the code, for coding decisions. That's valuable, and necessary for sound engineering, but we need user's documentation, as well.

**Comment by zigen** [comment]
yeah, my proposal for developers, not for users. my proposal should be addressed in other issue.

readthedocs seems good.
for example, [qulacs](http://docs.qulacs.org/en/latest/intro/4.1_python_tutorial.html) uses readthedocs and we can download epub.
![Screen Shot 2020-10-07 at 14 25 53](https://user-images.githubusercontent.com/3610296/95290798-1fbe6a00-08a9-11eb-942e-c020f75f74b2.png)

readthedocs provides [downloadable documentation](https://docs.readthedocs.io/en/stable/downloadable-documentation.html) feature. this seems what we need.

**Comment by icepolarizer** [comment]
Agreed. Also, the integration to GitHub seems to be smooth.

**Comment by wolfitis** [comment]
What I suggested was the general documentation, it should be easy enough so general users can follow. For developers it shouldn't be an issue to follow the same documentation for contributions. We might not need multiple docs for users and developers.

**Comment by rdviii** [comment]
It's Wednesday Japan time now...in 5.5 days, we will be doing a tutorial on QuISP for IEEE Quantum Week. Can a useful reorg of docs be completed in that time?  If not, we should either branch or plan to start work on it _after_ next week's tutorial.

**Comment by icepolarizer** [comment]
If you're talking about the conversion of `quisp/doc` into readthedocs, I think it's possible. Doxygen <-> Readthedocs? Not really.

**Comment by icepolarizer** [comment]
If you're talking about the conversion of quisp/doc into readthedocs, I think it's possible. 
Doxygen <-> Readthedocs? If then I can't make it in time.

**Comment by rdviii** [comment]
Thanks to @icepolarizer we now have a table of contents, which is a good start! See PR #177, merged into master.

**Comment by wolfitis** [comment]
The links are returning 404 error

**Comment by icepolarizer** [comment]
Just fixed. Thank you.
